tests/flytekit/unit/annotated/test_serialization.py

```diff
@@ -102,11 +118,47 @@ def my_wf(a: int) -> str:
     assert my_wf(a=2) == "hello"
 
     ctx = FlyteContext.current_context()
+    default_img = Image(name="default", fqn="test", tag="tag")
     registration_settings = context_manager.RegistrationSettings(
-        project="project", domain="domain", version="version", image="image", env=None,
+        project="project",
+        domain="domain",
+        version="version",
+        env=None,
+        image_config=ImageConfig(default_image=default_img, images=[default_img]),
     )
     with ctx.current_context().new_registration_settings(registration_settings=registration_settings):
         wf = my_wf.get_registerable_entity()
         assert wf is not None
         assert len(wf.nodes) == 2
         assert wf.nodes[1].branch_node is not None
+
+
+def test_serialization_images():
+    @task(container_image="{{.image.xyz.fqn}}:{{.image.default.version}}")
```

Comment:
can we add a couple tests with hardcoded strings as well?  like without the {{}}

Reply:
done!